### PR TITLE
feat: Finish Implementation of MPSideLoadedKits

### PR DESCRIPTION
**Pull request by BrandonStalnaker**
## Summary
 - Connected sideloaded kit filter configurations to mparticleoptions nad added unit test for all filters

 ## Testing Plan
 - Added unit tests and tested manually

 ## Reference Issue (For mParticle employees only.  Ignore if you are an outside contributor)
 - Closes https://go.mparticle.com/work/SQDSDKS-5345

### CODE REVIEW

File: UnitTests/MPKitContainerTests.m [hunk 1]
```diff
@@ -2554,6 +2581,507 @@ - (void)testAppInfoContainsSideloadKitsFlag {
...
+    NSSet<id<MPExtensionProtocol>> *registeredKits = [MPKitContainer registeredKits];
+    id registeredKit = [[registeredKits objectsPassingTest:^BOOL(id<MPExtensionProtocol>  _Nonnull obj, BOOL * _Nonnull stop) {
+        if ([obj conformsToProtocol:@protocol(MPExtensionKitProtocol)]) {
+            id<MPExtensionKitProtocol> kitExtension = (id<MPExtensionKitProtocol>)obj;
+            if (kitExtension.code.intValue == 1000000000) {
+                return YES;
+            }
+        }
+        return NO;
+    }] anyObject];
```
Comment (einsteinx2) [inline UnitTests/MPKitContainerTests.m:2606]:
```suggestion
    NSSet<id<MPExtensionProtocol>> *registeredKits = [MPKitContainer registeredKits];
    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"code == 1000000000"];
    id registeredKit = [[registeredKits filteredSetUsingPredicate:predicate] anyObject];
```

Seems like this does the same thing, but is much easier to read and less lines of code. I see it's used in the other tests further down so would be nice to standardize on it.

File: UnitTests/MPKitContainerTests.m [hunk 2]
```diff
@@ -2554,6 +2581,507 @@ - (void)testAppInfoContainsSideloadKitsFlag {
...
```
Comment (einsteinx2) [inline UnitTests/MPKitContainerTests.m:2638]:
```suggestion
    NSSet<id<MPExtensionProtocol>> *registeredKits = [MPKitContainer registeredKits];
    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"code == 1000000000"];
    id registeredKit = [[registeredKits filteredSetUsingPredicate:predicate] anyObject];
```

File: UnitTests/MPKitContainerTests.m [hunk 3]
```diff
@@ -2554,6 +2581,507 @@ - (void)testAppInfoContainsSideloadKitsFlag {
...
```
Comment (einsteinx2) [inline UnitTests/MPKitContainerTests.m:2665]:
```suggestion
    NSSet<id<MPExtensionProtocol>> *registeredKits = [MPKitContainer registeredKits];
    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"code == 1000000000"];
    id registeredKit = [[registeredKits filteredSetUsingPredicate:predicate] anyObject];
```

File: UnitTests/MPKitContainerTests.m [hunk 4]
```diff
@@ -2554,6 +2581,507 @@ - (void)testAppInfoContainsSideloadKitsFlag {
...
```
Comment (einsteinx2) [inline UnitTests/MPKitContainerTests.m:2700]:
```suggestion
    NSSet<id<MPExtensionProtocol>> *registeredKits = [MPKitContainer registeredKits];
    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"code == 1000000000"];
    id registeredKit = [[registeredKits filteredSetUsingPredicate:predicate] anyObject];
```

File: UnitTests/MPKitContainerTests.m [hunk 5]
```diff
@@ -2554,6 +2581,507 @@ - (void)testAppInfoContainsSideloadKitsFlag {
...
```
Comment (einsteinx2) [inline UnitTests/MPKitContainerTests.m:2730]:
```suggestion
    NSSet<id<MPExtensionProtocol>> *registeredKits = [MPKitContainer registeredKits];
    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"code == 1000000000"];
    id registeredKit = [[registeredKits filteredSetUsingPredicate:predicate] anyObject];
```

**Review by einsteinx2 — CHANGES_REQUESTED** [review]

**Review by einsteinx2 — APPROVED** [review]
lgtm thanks for the super comprehensive tests!!

File: UnitTests/MPKitContainerTests.m (hunk 1 above)
Comment (BrandonStalnaker) [reply UnitTests/MPKitContainerTests.m]: sorry about that I thought of a better way halfway through and didn't update the older tests

**Review by samdozor — COMMENTED** [review]
Looks good - approved for blackout.

**Pull request merged**